src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -48,36 +58,129 @@ public void run(Connection conn, List<Object> params) throws SQLException {
     public void run(Connection conn) throws SQLException {
         QueryTemplateInfo queryTemplateInfo = this.getQueryTemplateInfo();
 
-        try (PreparedStatement stmt = this.getPreparedStatement(conn, queryTemplateInfo.getQuery()); ResultSet rs = stmt.executeQuery()) {
+        try (PreparedStatement stmt = this.getPreparedStatement(conn, queryTemplateInfo.getQuery());
+                ResultSet rs = stmt.executeQuery()) {
             while (rs.next()) {
-                //do nothing
+                // do nothing
             }
         }
         conn.commit();
     }
 
-    public PreparedStatement getStatement(Connection conn, List<Object> params) throws SQLException {
+    public PreparedStatement getStatement(Connection conn, List<ComplexValue> params) throws SQLException {
         QueryTemplateInfo queryTemplateInfo = this.getQueryTemplateInfo();
 
         PreparedStatement stmt = this.getPreparedStatement(conn, queryTemplateInfo.getQuery());
         String[] paramsTypes = queryTemplateInfo.getParamsTypes();
         for (int i = 0; i < paramsTypes.length; i++) {
-            if (paramsTypes[i].equalsIgnoreCase("NULL")) {
+
+            ComplexValue param = params.get(i);
+            boolean hasDist = param.getDist().length() > 0;
+            boolean hasValue = param.getValue().length() > 0;
+
+            if ((!hasDist && !hasValue) || paramsTypes[i].equalsIgnoreCase("NULL")) {
                 stmt.setNull(i + 1, Types.NULL);
+            
+            } else if (hasDist) {
+                String distribution = param.getDist();
```

Comment:
@bpkroth I made some changes to the architecture to better support the distributions. Would this be something you feel like you could work with?

The new system would still take the <value> element but now allows to choose a distribution like this:
<value dist="uniform" min="0" max="10" seed="0" />
While still working with a single value: <value>10</value> and also NULL type: <value />
(So no breaking change)

So far I have only implemented the uniform distribution because I do not want to change everything again if there is feedback to the Architecture :)